### PR TITLE
Pull before export cli

### LINKED ISSUE #408: Add pull before export option to CLI operation
We have this option on the UI and we should bring that into the CLI as well. 

This would simplify any automatization scripts by removing the need of executing Briefcase for pulling the forms before exporting them.

Pulling from an Aggregate server requires hard-coding passwords on the CLI command. By removing those operations from such scripts we can make them more secure as well.

In order to benefit from this new option, users would make sure that:

1. They have selected the option to "Remember passwords" on the settings tab
2. After this, they have pulled at least once every form which they'd want to export on the CLI

The proposed CLI param would be `-pb` and `--pull_before`.

### PR DESCRIPTION
**Pull request by neoliteconsultant**
Closes #408 

#### Will require a change of documentation to include the new export cli option: --pull_before
https://github.com/opendatakit/docs/issues/768

### CODE REVIEW

File: src/org/opendatakit/briefcase/operations/Export.java [hunk 1]
```diff
@@ -83,10 +91,43 @@ public static void export(String storageDir, String formid, Path exportDir, Stri
...
         Optional.empty(),
         Optional.of(overwriteFiles)
     );
+
+
+    if (configuration.resolvePullBefore()) {
+      BriefcasePreferences appPreferences = BriefcasePreferences.appScoped();
+      FormStatus formStatus = new FormStatus(FormStatus.TransferType.EXPORT, formDefinition);
+
+      System.out.println("Pull before command specified for " + formid);
```
Comment (dcbriccetti) [inline src/org/opendatakit/briefcase/operations/Export.java]: Remove? Change to logging?

File: src/org/opendatakit/briefcase/ui/BriefcaseCLI.java [hunk 2]
```diff
@@ -55,6 +55,7 @@
   private static final String ODK_PASSWORD = "odk_password";
   private static final String ODK_USERNAME = "odk_username";
   private static final String OVERWRITE_CSV_EXPORT = "overwrite_csv_export";
+  private static final String PULL_BEFORE_EXPORT = "pull_before";
```
Comment (ggalmazor) [inline src/org/opendatakit/briefcase/ui/BriefcaseCLI.java]: Don't create the param here. This is the legacy Launcher and we want users to switch to the new Launcher. Let's use the new features to get them to the new Launcher.

File: src/org/opendatakit/briefcase/ui/BriefcaseCLI.java [hunk 3]
```diff
@@ -296,6 +297,7 @@ public void run() {
     // note that we invert incoming value
     boolean exportMedia = !cli.hasOption(EXCLUDE_MEDIA_EXPORT);
     boolean overwrite = cli.hasOption(OVERWRITE_CSV_EXPORT);
+    boolean pullBefore = cli.hasOption(PULL_BEFORE_EXPORT);
```
Comment (ggalmazor) [inline src/org/opendatakit/briefcase/ui/BriefcaseCLI.java]: Remove

File: src/org/opendatakit/briefcase/ui/BriefcaseCLI.java [hunk 4]
```diff
@@ -314,6 +316,7 @@ public void run() {
             fileName,
             exportMedia,
             overwrite,
+            pullBefore,
```
Comment (ggalmazor) [inline src/org/opendatakit/briefcase/ui/BriefcaseCLI.java]: Remove

File: src/org/opendatakit/briefcase/operations/Export.java (hunk 1 above)
Comment (neoliteconsultant) [reply src/org/opendatakit/briefcase/operations/Export.java]: Ok I will remove

File: src/org/opendatakit/briefcase/ui/BriefcaseCLI.java (hunk 2 above)
Comment (neoliteconsultant) [reply src/org/opendatakit/briefcase/ui/BriefcaseCLI.java]: ok

File: src/org/opendatakit/briefcase/ui/BriefcaseCLI.java [hunk 5]
```diff
@@ -316,7 +309,7 @@ public void run() {
             fileName,
             exportMedia,
             overwrite,
-            pullBefore,
+            false,
```
Comment (ggalmazor) [inline src/org/opendatakit/briefcase/ui/BriefcaseCLI.java]: Perfect, Thanks!

**Comment by ggalmazor** [comment]
It seems like you're having trouble with the checkstyle settings in your IDE. Are you using Intellij?

From the README.md:
> If you're using IntelliJ IDEA, we also recommend you [import the code style scheme](https://www.jetbrains.com/help/idea/copying-code-style-settings.html) for Briefcase at `config/codestyle/codestyle.xml`. Once you activate that scheme, use the automatic reformatting tool to produce code that will comply with the checkstyle rules of the project.

**Comment by ggalmazor** [comment]
@opendatakit-bot label "needs testing"

**Comment by neoliteconsultant** [comment]
@lognaturel , @yanokwa , and @dcbriccetti squash this commit to master

**Comment by kkrawczyk123** [comment]
**Tested with success!**
Verified on Ubuntu, MacOS and Windows. 
I suggest a little improvement: https://github.com/opendatakit/briefcase/issues/585

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"

**Pull request merged**